### ISSUE TITLE
bug: annotated type for openapi parameter examples

### ISSUE THREAD

**Issue by peterschutt** [litestar-org/litestar#2495]
Given a query parameter defined as:

```py
@get(path="/")
async def index(
    text: Annotated[
        str,
        Parameter(
            examples=[Example(value="example value", summary="example summary")]
        )
    ]
) -> str:
    return text
```

The examples render as objects, e.g, Elements:

![image](https://github.com/litestar-org/litestar/assets/20659309/1e64b9a2-aaa6-46de-9777-5390a9652f62)

Redoc:

![image](https://github.com/litestar-org/litestar/assets/20659309/f61e039b-62fd-47c5-8b3e-e2c9905af42c)

It seems to work better if the examples are passed as strings, but our types don't allow it:

![image](https://github.com/litestar-org/litestar/assets/20659309/8b8118c0-06f7-4c7e-b306-1bfcc88ce348)

![image](https://github.com/litestar-org/litestar/assets/20659309/b4139fdd-95a7-4091-b0f9-4cc5cdc0e183)

![image](https://github.com/litestar-org/litestar/assets/20659309/8982e304-bc91-49e3-bb69-0bb55fac7f89)
            

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh Litestar dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/2495">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2495/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/2495/pledge.svg">
</picture>
</a>

**Comment by Teraskull** [comment]
OpenAPI Spec for examples: https://spec.openapis.org/oas/latest.html#example-object-examples

**Comment by peterschutt** [comment]
Thanks @Teraskull - OK I'm slowly catching on.

This example:

```py
@get(path="/")
async def index(
    text: Annotated[
        str,
        Parameter(
            examples=[Example(value="example value", summary="example summary")]
        )
    ]
) -> str:
    return text
```
should look like this?

```py
@get(path="/")
async def index(
    text: Annotated[
        str,
        Parameter(
            examples={"an-example": Example(value="example value", summary="example summary")}
        )
    ]
) -> str:
    return text
```
And that mapping of example name to example object should exist outside of of the schema object, but we are rendering it as a schema example - which is why these are rendering funky or not at all? Does that sound like I've caught up?

If so, I'll reopen your other issue in favor of this one.

**Comment by Teraskull** [comment]
@peterschutt Yes, pretty much.

This is the only example format that renders properly on SwaggerUI and others. Litestar can probably generate the keys on it's own, since I don't think the key is used or shown anywhere on the UI.

So to not break compatibility, the `Example()` format can stay the same, and then the keys can be added dynamically.

**Comment by peterschutt** [comment]
> Litestar can probably generate the keys on it's own, since I don't think the key is used or shown anywhere on the UI.

OK, thanks for this info.

Closed in favor #2494 - thanks for persisting.
